### ISSUE TITLE
Add migration assistant for OSTicket

### ISSUE THREAD

**Issue by rolfschmidt** [zammad/zammad#354]
Zammad wish of Jonas Neumann:
"A Mirgration Assistant from other Ticket systems would be great! We are currently using OSTicket but your Product seems so much better - will it be possible to migrate all Old (and Active) Tickets to your system?"

**Comment by jacotec** [comment]
Good idea ... I'm currently also using OSTicket and have a close look to the Zammad development to switch over at some point.

**Comment by Taps7734** [comment]
Would definitely appreciate this as well.  I have just found this product and would be interested in seeing a migration from OSTicket.

**Comment by liangxinhui** [comment]
+1

**Comment by tlueder** [comment]
+1

**Comment by esxcx** [comment]
+1

**Comment by Zappelphilipp** [comment]
same here. would be awsome.

**Comment by schannall** [comment]
I'd like that too!

**Comment by schannall** [comment]
Hi,
is there any news on this? A workaround with an exported Database would be okay for me as well but I would not know where to start. Does Zammad have a CSV Import - this would lose our attachments but that would be okay for me..?

Regards, Jonas

**Comment by NMonst4** [comment]
+1

**Comment by altzone** [comment]
+1, good idea!

**Comment by thorsteneckel** [comment]
Locking conversation. There is no new input. +1 and requesting comments create noise and distraction. We are working hard on the next upcoming/most requested features and projects of our support customers. If you want to state your interest in the feature and give valuable input please give the initial post a 👍 reaction and we are able to evaluate it. We will lock all distracting issues from now on. Thanks for your understanding.

**Comment by thorsteneckel** [comment]
Hi there. We're wondering if there is someone using the [osTicket hosted service](supportsystem.com) and thinks about switching to Zammad. Do you have access to the data of your hosted system? Can you export it in any way? In which format will it be? We haven't found anything in their Knowledgebase or support forum...

We're currently evaluating creating an import but are not sure how to address all the use cases we have:

1.) osTicket on premise -> Zammad on premise
2.) osTicket on premise -> Zammad hosted
3.) osTicket hosted -> Zammad on premise
4.) osTicket hosted -> Zammad hosted

The most maintainable and straight forward way would be to directly connect to their MySQL database. This would work for use case 1. and maybe for 2. but not for 3. and 4. ...

Having a dedicated service/tool to provide a (REST) endpoint we could consume would introduce the maintenance overhead of a second application. Additionally that would require to setup that tool and having a compatible runtime environment. This will also not work for use case 3. and 4.

Having an osTicket plugin won't work either for use case 3. and 4. We'd also have to gain some domain knowledge of their codebase and are dependent on their internal API which will bring some maintainable difficulties over time.

**Comment by snizzleorg** [comment]
Case 1 or 2 would be enough for our installation

**Comment by thorsteneckel** [comment]
Please use the 👍 on @snizzleorg s comment if this applies to your use case.

**Comment by ggmanugg** [comment]
@thorsteneckel 
will there come an assistant for the described cases (1 and 2) or not in the future?

**Comment by thorsteneckel** [comment]
Thanks @ggmanugg for reaching out. We will provide a migrator as described for use case 1/2 which will not work with OSTicket hosted setup referenced as 3/4. However, the customer who asked us to give an estimation on the costs never replied so we have to postpone this for now :( However, we will pick it up as soon as we have the resources to do so. This might take some time.

I guess there won't be any additional information provided in this issue but questions on the state. This distracts us from from working on tasks - as e.g. this one. Therefore I'll lock this issue again. We'll get back to you as soon as we have something to share. Promise! ✌️
